addons/account/models/account_tax.py

```diff
@@ -225,6 +238,84 @@ def _compute_tax_group_id(self):
+                        'Tax Grids': repartition_line.tag_ids.mapped('name') or 'None',
+                        'Use in tax closing': str(repartition_line.use_in_tax_closing),
+                    }
+                    repartition_lines_str += str(repartition_line_info) + '//'
+                repartition_lines_str = repartition_lines_str.strip('//')
+            tax.repartition_lines_str = repartition_lines_str
+
+    def _message_log_repartition_lines(self, old_value_str, new_value_str):
+        old_values = old_value_str.split('//')
+        new_values = new_value_str.split('//')
+        if len(old_values) != len(new_values):
+            raise UserError(_("Taxes should have the same amount of invoice and refund repartition lines."))
+
+        kwargs = {}
+        for old_value, new_value in zip(old_values, new_values):
+            if old_value != new_value:
+                old_value = ast.literal_eval(old_value)
+                new_value = ast.literal_eval(new_value)
+                diff_keys = [key for key in old_value if old_value[key] != new_value[key]]
+                repartition_line = self.env['account.tax.repartition.line'].search([('id', '=', new_value['id'])])
+                body = "<b>{type}</b> repartition line {seq}:<ul class='mb-0 ps-4'>{changes}</ul>".format(
+                    type=repartition_line.document_type.capitalize(),
+                    seq=repartition_line.sequence + 1,
+                    changes=''.join(
+                        ["""
+                            <li>
+                                <span class='o-mail-Message-trackingOld me-1 px-1 text-muted fw-bold'>{old}</span>
+                                <i class='o-mail-Message-trackingSeparator fa fa-long-arrow-right mx-1 text-600'/>
+                                <span class='o-mail-Message-trackingNew me-1 fw-bold text-info'>{new}</span>
+                                <span class='o-mail-Message-trackingField ms-1 fst-italic text-muted'>({diff})</span>
+                            </li>""".format(old=escape(old_value[diff_key]), new=escape(new_value[diff_key]), diff=escape(_(diff_key)))
+                        for diff_key in diff_keys]
+                    )
+                )
+                kwargs['body'] = Markup(body)
+                super()._message_log(**kwargs)
+
+    def _message_log(self, **kwargs):
+        # OVERRIDE
+        self.ensure_one()
+
+        if self.is_used:
+            repartition_line_str_field_id = self.env['ir.model.fields'].search([('name', '=', 'repartition_lines_str')], limit=1).id
+            for tracked_value_id in kwargs['tracking_value_ids']:
+                if tracked_value_id[2]['field_id'] == repartition_line_str_field_id:
+                    kwargs['tracking_value_ids'].remove(tracked_value_id)
+                    self._message_log_repartition_lines(tracked_value_id[2]['old_value_char'], tracked_value_id[2]['new_value_char'])
+
+            return super()._message_log(**kwargs)
```

Comment:
this method needs documentation, and ho, look... there is an OVERRIDE comment that you could use :)